### ISSUE TITLE
revision for GO.xrf_abbs

### ISSUE THREAD

**Issue by murphyte** [geneontology/helpdesk#90]
Hi GO,

We occasionally get questions about HGNC and MGI dbxrefs, and a user noticed that the description in the GO.xrf_abbs file isn't quite correct:
ftp://ftp.geneontology.org/pub/go/doc/GO.xrf_abbs

For both HGNC and MGI, the *id* itself starts with "HGNC:" or "MGI:". That is, their IDs are NOT integers, but strings that include a prefix (similar to FlyBase FBgn###, but with a colon). So a properly formatted dbxref of the format "tag:id" is HGNC:HGNC:29 or MGI:MGI:97490 (because HGNC: or MGI: is BOTH the database tag, AND part of the ID itself). We had a request from both to display db_xrefs that way in INSDC files, so it would make sense to document it the same way in the GO.xrf_abbs file. As silly as it looks, it does have the effect of anyone parsing on the first colon will properly divide the string up into a tag (which corresponds to the base URL), and the actual ID string that HGNC and MGI intend you to use, and help prevent thinking the ID is just an integer. 

Best regards,
-Terence

**Comment by kltm** [comment]
@murphyte 

To clarify a little about where our metadata comes from, the upstream file that we use to organize the xrefs is: https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml. There is a process somewhere that converts that to the legacy GO.xrf_abbs format, but I'd have to poke around a little to dig that out.

That said, while MGI does have the "double" namespacing documented (https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml#L1729), HGNC does not and seems to be resolvable/usable as we have it with the local id not including the additional namespace reference (https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml#L1209). Could you possibly clarify what the expected contents of the HGNC entry would be?

(tagging @cmungall)

**Comment by murphyte** [comment]
Thanks, I guess I didn't actually check the MGI entry. The user mail that prompted this ticket was specifically asking about HGNC.

The HGNC site is more flexible in terms of working both with and without the prefix in their URLs, whereas the MGI site requires it.  But back when this first came up HGNC said including the "HGNC:" in the ID was their preferred format.

For your HGNC entry, it would just need a change to example_id and example_URL:
      url_syntax: http://www.genenames.org/data/hgnc_data.php?hgnc_id=HGNC:[example_id]
      example_id: HGNC:HGNC:29
example_url: http://www.genenames.org/data/hgnc_data.php?hgnc_id=HGNC:HGNC:29

Thanks!

-Terence

**Comment by cmungall** [comment]
Hi Terence,

All identifiers in the GO xrefs file are [CURIEs](https://en.wikipedia.org/wiki/CURIE), double `:` is syntactically forbidden. The MGI entry is actually a historic mistake, we will be switching to CURIEs soon. We always include the `:` as part of the ID (aka CURIE aka global ID). So for example we treat all the following as IDs

 * HGNC:1
 * FlyBase:FBgn1
 * GO:0000001

We treat the part after the `:` as the *local id*. Sometimes these redundantly hint at the source (as in FBgn) but in other cases they may be purely numeric.

The fields we use in the yaml are a bit confusing. Every instance of `example_id` should really say  `example_local_id` but this would cause too much churn to change, and it would have no effect on the behavior. In this case we store the ID as `HGNC:nnnn` so the current entry is correct for creating resolvable URLs.

We want to respect individual sources wishes as to how their IDs are displayed (e.g. FBgnNN vs FlyBase:FBgnNNN), we have a separate ticket for this. For now, all sources are shown as the global ID.

See also: http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2001414

**Comment by murphyte** [comment]
I mostly look at that GO.xrf_abbs file because it's referenced in the GFF3 spec. I'm not sure if that's how the user who prompted this ticket was finding the file, but it seems likely. The GFF3 spec doesn't prohibit the double colon, and neither does the INSDC flatfile spec.

When HGNC and MGI first raised the issue with their IDs we tried quite hard to talk them out of it without success. They were both quite adamant that what you are referring to as the _local id_ also includes the HGNC: or MGI: prefix, which equates to a dbxref as spec'd in either INSDC flatfile or SO GFF3 format being written as HGNC:HGNC:1. I also dislike the extra colon, but they both maintained that it was part of their local ID formats since day one. You're welcome to bring it up with them again, but at this point I wouldn't be particularly enthusiastic about changing given we already had to go through one pretty extensive data retro and code review to accommodate their change request in the first place.

-Terence
